src/ansys/dpf/post/data_object.py

```diff
@@ -66,38 +67,61 @@ def as_data_frame(self, columns=None, **kwargs):
         >>> import pandas as pd
         >>> from ansys.dpf import post
         >>> from ansys.dpf.post import examples
-        >>> solution = post.load_solution(examples.multishells_rst, legacy=False)
-        >>> #stress = solution.stress()
-        >>> #stress.xx.plot_contour(show_edges=False)
+        >>> solution = post.load_simulation(examples.multishells_rst)
+        >>> # Export the displacements vector field at step 1 as a DataFrame
+        >>> displacement = solution.displacement(steps=[1], nodes=[1, 2, 3])
+        >>> df = displacement.as_data_frame()
+        >>> print(df)
+                  X          Y         Z
+        1  0.398320 -13.797378 -0.163767
+        2  0.233114 -13.797652 -0.153190
+        3  0.367542 -13.808151 -0.163739
+        >>> print(df.name)
+        displacement_1.s
 
-        >>> #data = stress.xx.get_data_at_field(0)
-        >>> #data_frame = pd.DataFrame(data)
-
-        >>> #data2 = [[0,1,2], [3,4,5], [6,7,8], [9,10,11]]
-        >>> #columns = ['x', 'y', 'z']
-        >>> #df = pd.DataFrame(data2, columns)
-        >>> #df = df.transpose()
         """
         try:
             import pandas as pd
         except ModuleNotFoundError:
             raise PandasImportError
-        # columns = None
-        # for arg in args:
-        #     columns.appends(arg)
 
-        # #load data into a DataFrame object:
-        # data_frame = pd.DataFrame(self.field(0), columns)
-        # transposed = data_frame.transpose()
         if not columns:
             columns = self._columns
-        return pd.DataFrame(
+        # The wrapped FieldsContainer should only contain one Field
+        if len(self) > 1:
+            raise NotImplementedError(
+                "This function requires for the DataObject to contain only"
+                "one data entity."
+            )
+        df = pd.DataFrame(
             self.as_array(), columns=columns, index=self._mesh_scoping.ids
         )
+        df.name = self._fc[-1].name
+        return df
 
     def as_array(self):
-        """Return the DataObject as a NumPy array."""
-        return np.array(self._fc[-1].data)
+        """Return the DataObject as a NumPy array.
+
+        Returns
+        -------
+        class:`pandas.core.frame.DataFrame`
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> from ansys.dpf import post
+        >>> from ansys.dpf.post import examples
+        >>> solution = post.load_simulation(examples.multishells_rst)
+        >>> # Export the displacements vector field at step 1 as a DataFrame
+        >>> displacement = solution.displacement(steps=[1], nodes=[1, 2, 3])
+        >>> arr = displacement.as_array()
+        >>> print(arr)
+        [[  0.39831985 -13.79737819  -0.16376683]
+         [  0.23311378 -13.79765179  -0.15318972]
+         [  0.36754181 -13.80815135  -0.16373945]]
+
+        """
+        return self._fc[-1].data
```

Comment:
I think returning the last field should be fine for most cases (static, transient mechanical, fluids, etc). But for modal analysis we should return the 1st mode. I am not sure if that is the only case that is different tho.

Reply:
> I think returning the last field should be fine for most cases (static, transient mechanical, fluids, etc). But for modal analysis we should return the 1st mode. I am not sure if that is the only case that is different tho.

@GuillemBarroso You are right, I'll have to change that behavior once I implement the "ModalSimulation" type or something like that. The DataObject should thus have an idea of the type of simulation it is coming from.

Reply:
Great!